src/classes/ADDR_Contact_TDTM.cls

```diff
@@ -50,168 +51,195 @@ public class ADDR_Contact_TDTM extends TDTM_Runnable {
+	        if (listConAddrSet.size() > 0) {
+	            setConAddrFromLookup(listConAddrSet);
+	            // to avoid pushing these address changes to the HH.
+	            ADDR_Addresses_TDTM.hasRunAddrTrigger = true;
+	        }
+    
+	        // update the Latest date fields on any address overrides
+	        if (mapAddrIdIsOverride.size() > 0) 
+	            ADDR_Addresses_UTIL.updateAddrIsOverride(mapAddrIdIsOverride, dmlWrapper);
+	            
+	        // create any new Default Address objects
+	        if (listConAddrCreate.size() > 0)
+	            createAddrFromContact(listConAddrCreate, dmlWrapper);                              
+        }
+        return dmlWrapper;
     }
-
+    
+    /*******************************************************************************************************
+    * @description for each Contact, create a new default address and add it to dmlWrapper
+    * @param Listcontact a List of Contacts
+    * @param dmlWrapper to hold the Addresses that need creating
+    * @return void
+    ********************************************************************************************************/
+    private void createAddrFromContact(List<Contact> listContact, DmlWrapper dmlWrapper) {
+        List<Address__c> listAddr = new List<Address__c>();
+        for (Contact contact : listContact) {
+            Address__c addr = new Address__c();
+            addr.Parent_Contact__c = contact.Id;
+            addr.Default_Address__c = true;
+            addr.Latest_Start_Date__c = system.today();
+            addr.Latest_End_Date__c = null;
+            ADDR_Addresses_UTIL.copyAddressStdSObjAddr(contact, 'Mailing', addr, null);            
+            ListAddr.add(addr);
+            UTIL_Debug.debug('****New address record: ' + JSON.serializePretty(addr));
+        }
+        //since coming from an Contact address, there is no Address Type, so exclude it from the match testing.
+        List<Address__c> newAddrs = ADDR_Addresses_UTIL.createNonDuplicateAddresses(ListAddr, dmlWrapper, false);
+        
+        //Match each Contact with each new address
+        List<ID> contactIDsToUpdate = new List<ID>();
+        Map<ID, Address__c> parentcontactIdToAddrMap = new Map<ID, Address__c>();
+        for(Address__c addr : newAddrs) {
+            if(addr.Parent_Contact__c != null) {
+               contactIDsToUpdate.add(addr.Parent_Contact__c);
+               parentcontactIdToAddrMap.put(addr.Parent_Contact__c, addr);  
+            }
+        }
+        
+        //We need to re-query before we are in the "after" part of the trigger. We cannot change it to before because
+        //we need the Contact ID to have a value when populating the addr.Parent_Contact__c field above.
+        List<Contact> parentContacts = [select is_Address_Override__c, Current_Address__c from Contact where ID in :contactIDsToUpdate];
+        List<Contact> ContactsToUpdate = new List<Contact>();
```

Comment:
Variables should use the `lowerCamelCase` naming convention. Please rename accordingly.